kratos/geometries/coupling_geometry.h

```diff
@@ -0,0 +1,357 @@
+ */
+template<class TPointType> class CouplingGeometry
+    : public Geometry<TPointType>
+{
+public:
+    ///@}
+    ///@name Type Definitions
+    ///@{
+
+    /**
+     * Geometry as base class.
+     */
+    typedef Geometry<TPointType> BaseType;
+    typedef Geometry<TPointType> GeometryType;
+
+    typedef typename GeometryType::Pointer GeometryPointer;
+
+    /**
+     * Pointer definition of CouplingGeometry
+     */
+    KRATOS_CLASS_POINTER_DEFINITION( CouplingGeometry );
+
+    typedef TPointType PointType;
+
+    typedef typename BaseType::IndexType IndexType;
+    typedef typename BaseType::SizeType SizeType;
+
+    typedef typename BaseType::PointsArrayType PointsArrayType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for coupling one master to one slave geometry.
+    CouplingGeometry(
+        GeometryPointer pMasterGeometry,
+        GeometryPointer pSlaveGeometry)
+        : BaseType(PointsArrayType(), &(pMasterGeometry->GetGeometryData()))
+    {
+        KRATOS_DEBUG_ERROR_IF(pMasterGeometry->Dimension() != pSlaveGeometry->Dimension())
+            << "Geometries of different dimensional size!" << std::endl;
+
+        mpGeometries.resize(2);
+
+        mpGeometries[0] = pMasterGeometry;
+        mpGeometries[1] = pSlaveGeometry;
+    }
+
+    explicit CouplingGeometry(const PointsArrayType& ThisPoints)
+        : BaseType()
+    {
+    }
+
+    /**
+     * Copy constructor.
+     * Construct this geometry as a copy of given geometry.
+     *
+     * @note This copy constructor does not copy the points and new
+     * geometry shares points with given source geometry. It is
+     * obvious that any change to this new geometry's point affect
+     * source geometry's points too.
+     */
+    CouplingGeometry( CouplingGeometry const& rOther )
+        : BaseType( rOther )
+    {
+    }
+
+    /**
+     * Copy constructor from a geometry with other point type.
+     * Construct this geometry as a copy of given geometry which
+     * has different type of points. The given goemetry's
+     * TOtherPointType* must be implicity convertible to this
+     * geometry PointType.
+     *
+     * @note This copy constructor does not copy the points and new
+     * geometry shares points with given source geometry. It is
+     * obvious that any change to this new geometry's point affect
+     * source geometry's points too.
+     */
+    template<class TOtherPointType> explicit CouplingGeometry(
+        CouplingGeometry<TOtherPointType> const& rOther )
+        : BaseType( rOther )
+    {
+    }
+
+    /**
+     * Destructor. Does nothing!!!
+     */
+    ~CouplingGeometry() override {}
+
+
+    GeometryData::KratosGeometryFamily GetGeometryFamily() const override
+    {
+        return GeometryData::Kratos_generic_family;
+    }
+
+    GeometryData::KratosGeometryType GetGeometryType() const override
+    {
+        return GeometryData::Kratos_generic_type;
```

Comment:
is it necessary to introduce another type?
Just asking, not sure if necessary

Reply:
You're right. It just overrides with the same. I will remove it.